### PR TITLE
Bug 1994139: Bump OLM kubernetes version to 1.22

### PR DESCRIPTION
**Pull request by exdx**
PR updates dependencies to kubernetes 1.22 for OLM and includes a fix for the `make manifests` directive to make it crossplatform. 

Signed-off-by: Daniel Sover dsover@redhat.com

Upstream-repository: operator-lifecycle-manager
Upstream-commit: 4a784fceec7661e36281139c4fc0f28a8e260c7f

Upstream-repository: api
Upstream-commit: 74543ec8c395244fa7892466b310ec38ec2ab375

Upstream-repository: operator-registry
Upstream-commit: a4fd56c7fb85428c6fb20d8e9fa378d3b5647657

### CODE REVIEW

File: scripts/generate_crds_manifests.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ CONTROLLER_GEN="go run ./vendor/sigs.k8s.io/controller-tools/cmd/controller-gen"
 HELM="go run helm.sh/helm/v3/cmd/helm"
 
 ver=$(cat ./staging/operator-lifecycle-manager/OLM_VERSION)
-tmpdir="$(mktemp -p . -d 2>/dev/null || mktemp -p . -d -t tmpdir)"
+tmpdir="$(mktemp -p . -d 2>/dev/null || mktemp -d ./tmpdir.XXXXXXX)"
```
Comment (timflannagan) [inline scripts/generate_crds_manifests.sh:15]: Let's move these changes to their own commit.

Comment (exdx) [reply scripts/generate_crds_manifests.sh:15]: Agreed

**Comment by exdx** [comment]
Closing in favor of #170
